File: Scraper/Scraper/Scraper.Core/Model/Match.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Scraper.Core.Model
{
    public class Match
    {
        [Key]
        public int Id { get; set; }
        public int MatchId { get; set; }
        public int? ParentId { get; set; }
        public int? EventId { get; set; }
        public int RoundId { get; set; }
        public string MatchName { get; set; }
        public DateTime MatchDate { get; set; }
        public DateTime LastUpdated { get; set; }
        public string Option1 { get; set; }
        public string Option2 { get; set; }
        public string Option3 { get; set; }
        public double Option1Odds { get; set; }
        public double Option2Odds { get; set; }
        public double Option3Odds { get; set; }
        public Result Result { get; set; }


        public bool HasParent => ParentId != null && EventId != null;

        public override string ToString()
        {
            return $"MatchDate: {MatchDate} - MatchId: {MatchId} - Options: {Option1} {Option2} {Option3} - Odds: {Option1Odds} {Option2Odds} {Option3Odds}";
        }
    }
}

File: Backend/models/match.js
module.exports = function (sequelize, DataTypes) {
  const Match = sequelize.define("matches", {
    Id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      allowNull: false,
      autoIncrement: true
    },
    MatchId: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    ParentId: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    EventId: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    RoundId: {
      type: DataTypes.INTEGER,
      allowNull: true
    },
    MatchName: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    MatchDate: {
      type: DataTypes.DATE,
      allowNull: true
    },
    LastUpdated: {
      type: DataTypes.DATE,
      allowNull: true
    },
    Option1: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    Option2: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    Option3: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    Option1Odds: {
      type: DataTypes.DOUBLE,
      allowNull: true
    },
    Option2Odds: {
      type: DataTypes.DOUBLE,
      allowNull: true
    },
    Option3Odds: {
      type: DataTypes.DOUBLE,
      allowNull: true
    },
    Missing: {
      type: DataTypes.BOOLEAN,
      defaultValue: false,
      allowNull: false
    },
    Invalid: {
      type: DataTypes.BOOLEAN,
      defaultValue: false,
      allowNull: false
    }
  });

  Match.associate = (models) => {
    Match.hasOne(models.results);
    Match.hasMany(models.bets);
  }

  return Match;
}